doc/release/1.11.0-notes.rst

```diff
@@ -41,8 +41,21 @@ Future Changes
 The following changes are scheduled for Numpy 1.12.0.
 
 * Support for Python 2.6, 3.2, and 3.3 will be dropped.
-* Slicing a ``MaskedArray`` will return views of both data **and** mask.
-  Currently the mask is returned as a copy.
+* When setting items of a ``MaskedArray`` using the square bracket
+  notation NumPy will in the future never copy the mask. Some
+  examples of this are ``child = ma_array[::2]; view[3] = 3``, where
+  the mask of ``child`` is a view into the mask of ``ma_array``.
+  Also when creating ``ma_array = MaskedArray(data, mask)`` will
+  make ``ma_array`` view the data in ``mask`` and setting an item
+  will currently copy ``mask`` but in the future modify the original.
```

Comment:
Few issues with this text:
* it's in the wrong place --it's currently under "things that will happen in 1.12", but there's no way we're going to flip the switch on this in 1.12, it's way too early.
* the description here is hard to understand i think. "Setting items of the array using the square bracket notation" to me sounds like `__setitem__`. I'm not 100% sure what the part about `MaskedArray` is saying either. Maybe: "Currently, taking a view of a masked array produces a confusing result. For example, if we write `masked_view = masked_original[:]`, then `masked_view`'s data array will be a view of `masked_original`'s data array, so modifications to one array's data will also affect the other: `masked_view[0] = 123; assert masked_original[0] == 123`. But currently, the *mask* array is copied, rather than taking a view, so changes to one array's mask will not affect the other: `masked_view[0] = np.ma.masked; assert masked_original[0] is not np.ma.masked`. A similar situation happens when explicitly constructing a masked array using `MaskedArray(data, mask)` -- the returned array will have a view of `data` but a copy of `mask`. In the future, these cases will be normalized so that the data and mask arrays are treated the same way, and modifications to either will propagate between views. In 1.11, numpy will issue a `whatever class it is` warning whenever user code modifies the mask of a view. To silence these warnings, and make your code robust against the upcoming changes, you have two options: if you want to keep the current behavior, call `masked_view.unshare_mask()` before modifying the mask. If you want to get the future behavior early, do `masked_view._unshare_mask = False`."